mantidproject/mantid
# Change Algorithm State tracking

**Description of work.**
Previously we had a collection of boolean values that tracked the state of an algorithm object:
1. IsInitialized
1. IsRunning
1. IsExecuted

This however cannot describe some situations, for example when an algorithm is errored.

This PR puts in two enums to describe the execution status, and the status of the result.

The boolean is methods are left in place, but report from the two enum states.

**To test:**

1. Run algorithms in both workbench and plot and check the Algorithm Progress monitors still work (the details buttons that bring up the dialog boxes).
1. Code review

re #28093. 

*This does not require release notes* because **it is purely internal**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
